src/conventions/h-group/clue-interpretation/connecting-cards.js

```diff
@@ -159,17 +160,82 @@ function find_unknown_connecting(game, giver, target, reacting, identity, connec
 				logger.warn(`disallowed hidden finesse on ${logCard(finesse)} ${finesse.order}, true ${logCard(identity)} could be duplicated in giver's hand`);
 				return;
 			}
 
-			return { type: 'finesse', reacting, card: finesse, hidden: true, identities: [finesse.raw()] };
+			return { type: 'finesse', reacting, card: finesse, hidden: true, bluff, identities: [finesse.raw()] };
 		}
 	}
 }
 
+/**
+ * Determines whether a bluff connection is a valid bluff, and updates the connection accordingly.
+ * @param {Game} game
+ * @param {number} giver 				The player index that gave the clue. They cannot deduce unknown information about their own hand.
+ * @param {number} target 				The player index receiving the clue. They will not find self-prompts or self-finesses.
+ * @param {Connection[]} connections	The complete connections leading to the play of a card.
+ * @param {ActualCard[]} promised		The non-hidden play(s) which are promised.
+ * @returns {Connection[]}
+ */
+export function resolve_bluff(game, giver, target, connections, promised) {
+	if (connections.length == 0 || !connections[0].bluff)
+		return connections;
+	const { state } = game;
+	const bluffCard = connections[0].card;
+	// Determine the next play if this is a bluff.
+	const next_play = connections.findIndex(connection => connection.card.order == promised[0].order) + 1;
+	// A bluff must be followed only by prompts.
+	let bluff_possible = connections.every((conn, index) => index < next_play || (!conn.hidden && conn.type !== 'finesse'));
+
+	if (bluff_possible) {
+		// A bluff must be recognizable. As such, there should be no connection
+		// between the bluffed card and at least one of the following plays
+		// as known by the player who would play next after the bluff play.
+		let known_bluff = false;
+		const next_player = state.hands.findIndex(hand => hand.find(c => c.order == promised[1].order));
+		const bluff_identies = (giver + 1) % state.numPlayers == state.ourPlayerIndex ?
+			game.players[state.ourPlayerIndex].thoughts[bluffCard.order].inferred.filter(c => state.isPlayable(c)) :
+			[bluffCard];
+		for (let i = 1; i < promised.length; ++i) {
+			const known_bluff_cards = bluff_identies.filter(identity => {
+				const expected = { suitIndex: identity.suitIndex , rank: identity.rank + i };
+				if (state.hands[next_player].find(c => c.order == promised[i].order)) {
+					const thoughts = game.players[next_player].thoughts[promised[i].order];
+					return !thoughts.inferred.has(expected);
+				} else {
+					return !promised[i].matches(expected);
+				}
+			});
+			if (known_bluff_cards.length > 0) {
+				known_bluff = true;
+				break;
+			}
+		}
+		bluff_possible = bluff_possible && known_bluff;
+	}
+
+
+	if (!bluff_possible) {
+		// If a bluff is not possible, we only have a valid connection if the real card was found.
+		// If it is in our hand, we have to assume it is the promised card.
+		if (connections[0].card == promised[0] && connections[0].hidden)
```

Comment:
If you're changing to compare cards using order, then there's another one here. :slightly_smiling_face:

Throughout your PR, either way should be fine since all of this is done at the same time. What can occur in other places is that the game gets cloned during a rewind/hypothetical, and then two objects don't have the same reference anymore. Using the order doesn't have this problem, so I've just gotten used to doing it that way.